File: 3334-AppleRedistributionIntoBoxes/3334-AppleRedistributionIntoBoxes.py
# Last updated: 6/11/2026, 11:18:52 AM
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        capacity.sort()
        sum1 =0;
        sum2 = sum(apple)
        curr = capacity[-1];
        sum1 += curr
        ct = 1
        while(sum1<sum2):
            capacity.remove(curr)
            curr = capacity[-1];
            sum1 += curr
            ct+=1
        return ct

File: 3334-AppleRedistributionIntoBoxes/3334-AppleRedistributionIntoBoxes.cpp
// Last updated: 6/11/2026, 11:18:47 AM
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& cap) {
        int s = accumulate(apple.begin(), apple.end(), 0);
        sort(cap.begin(), cap.end());
        int nob = 0;
        for(int i=cap.size()-1; i>=0; i--){
            nob++;
            s-=cap[i];
            if(s<=0) return nob;
        }

        return -2;
    }
};